test/jdk/tools/jpackage/windows/WinDirChooserTest.java

```diff
@@ -41,6 +41,7 @@
  * @build jdk.jpackage.test.*
  * @requires (os.family == "windows")
  * @modules jdk.jpackage/jdk.jpackage.internal
+ * @compile WinDirChooserTest.java
```

Comment:
Changes like this one is not necessary. `@run` will build WinDirChooserTest automatically.

`@compile` should be added only when necessary, like WinInstallerUiTest.java

Reply:
It is better to have `@compile` everywhere for consistency. 

The proper way to run tests is by passing test class name as an argument for test runner, test class should not have `main()` and should have `@Test` annotation for test methods. Some tests have not been updated to follow this design and adding `@compile` to them doesn't make a difference now. But eventually they will be updated and they will need `@compile` anyways.

Reply:
If this is the plan please update the bug description to reflect this. Otherwise it's not clear why this is done. The `@compile` in those files aren't really "missing" since they are not required. I would suggest changing the bug title to "Add @compile tags to jpackage tests"

Reply:
> adding `@compile` to them doesn't make a difference now

Also, the above is not true. `@compile` will always run javac every time you run the test. I think you should use `@build` instead -- if the class doesn't exist, it will be compiled. If it already exists (you run jtreg a second time), the class won't be compiled again, so it will run a little faster.

Reply:
> @compile will always run javac every time you run the test
Didn't know that. Thank you for explanation. Would this be OK to leave `@compile` tags?

Reply:
> > @compile will always run javac every time you run the test
> > Didn't know that. Thank you for explanation. Would this be OK to leave `@compile` tags?

Since you're making changes, I think we should do it correctly. We should use `@build` tags.